### ISSUE TITLE
MP4Sampler High CPU usage

### ISSUE THREAD

**Issue by jonicong** [shogo4405/lf.swift#129]
Hi, 

Current Spec:
- lf.swift v0.5.1
- Stream with rtmp://

When we're using "RTMPStream",  we found that CPU usage is incredible High (reach up to 100%) during streaming.

I think this problem caused by MP4Sampler and i don't know what this use for, please elaborate this thing :)

Also, when we stop the stream, CPU also remaining HIGH. I found there have some infinite loop in the logic below:

// MP4Sampler.swift 

```
extension MP4Sampler: Runnable {
    // MARK: Runnable
    final func startRunning() {
        lockQueue.async {
            self.running = true
            while (self.running) {
                self.fileQueue.sync {
                    self.run()  // ---> always loop here
                }
            }
        }
    }

    final func stopRunning() {
        lockQueue.async {
            self.running = false   // ---> when call stopRunning, never go to here, thread issue?
        }
    }
}
```

So, current my solution is just comment this code to solve this issue:
**sampler.startRunning()**  // RTMPStream.swift line 245

Thanks.

**Comment by denizak** [comment]
i wonder, what is the sampler used for?

**Comment by crossle** [comment]
while true cause the CPU 100%, u can add usleep()

**Comment by shogo4405** [comment]
Released 0.5.2
